$NetBSD: patch-ao,v 1.1 2012/07/31 08:16:07 he Exp $

Add Parrot_get_cpu_type() from generic/, since we have a local
misc.c for netbsd.

--- src/platform/netbsd/misc.c.orig	2012-06-06 18:03:39.000000000 +0000
+++ src/platform/netbsd/misc.c
@@ -22,6 +22,10 @@ Miscellaneous helper functions that are 
 
 #include "parrot/parrot.h"
 
+#ifdef PARROT_HAS_HEADER_SYSUTSNAME
+#  include <sys/utsname.h>
+#endif
+
 /* HEADERIZER HFILE: none */
 
 /*
@@ -53,6 +57,33 @@ Parrot_platform_init_code(void)
 
 /*
 
+=item C<STRING *Parrot_get_cpu_type(Parrot_Interp)>
+
+Fetch CPU type for non-win32 systems
+For win32, look in platform/win32/misc.c
+
+=back
+
+=cut
+
+*/
+
+STRING *
+Parrot_get_cpu_type(Parrot_Interp interp) {
+    struct utsname uname_info;
+    char  *proc_arch = "";
+
+#ifdef PARROT_HAS_HEADER_SYSUTSNAME
+    uname(&uname_info);
+    proc_arch = uname_info.machine;
+#endif
+    return Parrot_str_new_init(interp, proc_arch, strlen(proc_arch),
+            Parrot_ascii_encoding_ptr, 0);
+
+}
+
+/*
+
 =back
 
 =cut
